include/pokeathlon/pokeathlon.h

```diff
@@ -0,0 +1,91 @@
+#ifndef POKEHEARTGOLD_POKEATHLON_H
+#define POKEHEARTGOLD_POKEATHLON_H
+
+#include "overlay_manager.h"
+
+// Forward declarations
+typedef struct PokeathlonCourseWork PokeathlonCourseWork;
+typedef struct PokeathlonCourseArgs PokeathlonCourseArgs;
+
+// Pokeathlon course arguments structure
+struct PokeathlonCourseArgs {
+    u8 filler_0[4]; // 0x000
+    u32 field_4;    // 0x004 - Mode flag (1 = special mode)
+    // ... more fields to be discovered
+};
+
+// Pokeathlon course work structure (0xD70 = 3440 bytes)
+struct PokeathlonCourseWork {
+    u8 data[0xD70]; // Complete work structure (will be refined during decompilation)
+};
+
+// Helper inline functions to access work structure fields
+// These will be replaced with proper struct members as we discover the layout
+static inline u32 *PokeathlonWork_GetHeapIdPtr(PokeathlonCourseWork *work) {
```

Comment:
generally it's a good idea to fill in members as you come across them, it makes it cleaner, the members between can just be u8 arrays of padding